packages/NodeTypeResolver/PHPStan/Scope/Contract/NodeVisitor/NodeVisitorInterface.php

```diff
@@ -0,0 +1,11 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\NodeTypeResolver\PHPStan\Scope\Contract\NodeVisitor;
+
+use PhpParser\NodeVisitor;
+
+interface NodeVisitorInterface extends NodeVisitor
```

Comment:
to ease differentiating `NodeVisitorInterface` from `NodeVisitor` it might make sense to use a more specific name for `NodeVisitorInterface`

Reply:
What interface name do you think better?

Reply:
maybe `NodeScopeVisitor`

Reply:
or `ScopeNodesVisitor`

Reply:
naming is hard, but I guess the lengthy name is `PHPStanNodeScopeResolverNodeVisitorInterface`, as they are used in `PHPStanNodeScopeResolver`.
up to you what can be removed.
maybe `ScopeResolverNodeVisitorInterface` or shorter `ScopeResolverVisitorInterface`?

Reply:
I like `ScopeResolverNodeVisitorInterface` 👍

Reply:
I [renamed interface to ScopeResolverNodeVisitorInterface](https://github.com/rectorphp/rector-src/pull/3831/commits/5cca0a44aacab8c3c3939aeef66cf49f1acb696c)